leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/EndpointContextUtil.java

```diff
@@ -30,12 +31,19 @@
 import org.eclipse.californium.elements.auth.PreSharedKeyIdentity;
 import org.eclipse.californium.elements.auth.RawPublicKeyIdentity;
 import org.eclipse.californium.elements.auth.X509CertPath;
+import org.eclipse.californium.oscore.OSCoreEndpointContextInfo;
 import org.eclipse.leshan.core.request.Identity;
 
 public class EndpointContextUtil {
 
     public static Identity extractIdentity(EndpointContext context) {
         InetSocketAddress peerAddress = context.getPeerAddress();
+        // Build identity for OSCORE if it is used
+        if (context.get(OSCoreEndpointContextInfo.OSCORE_SENDER_ID) != null) {
+            String oscoreIdentity = "sid=" + context.get(OSCoreEndpointContextInfo.OSCORE_SENDER_ID) + ",rid="
+                    + context.get(OSCoreEndpointContextInfo.OSCORE_RECIPIENT_ID);
+            return Identity.oscoreOnly(peerAddress, oscoreIdentity.toLowerCase());
+        }
```

Comment:
This should be done in an `else` block of the code above. (senderIdentity == null)